scio-core/src/main/scala/com/spotify/scio/ScioContext.scala

```diff
@@ -510,6 +511,13 @@ class ScioContext private[scio] (
     o.setScioVersion(BuildInfo.version)
   }
 
+  private[scio] def labels: Map[String, String] =
+    (for {
+      // Check if class is in classpath
+      _ <- Try(Class.forName("org.apache.beam.runners.dataflow.options.DataflowPipelineOptions"))
+      o <- Try(optionsAs[DataflowPipelineOptions])
```

Comment:
This line should be an no-op if the one above fails right? So it shouldn't cause a `ClassNotFoundException` if DF runner deps are absent?

Reply:
Yes that's exactly what it does :)